boards/common/wsn430/board_init.c

```diff
@@ -15,111 +15,102 @@
-    P3OUT = 0x00;    // Port3 Output: 00000000 = 0x00
-    P3DIR = 0xFF;    // Port3 Direction: 11111111 = 0xFF
+    /* Port 3: UART */
+    P3SEL = 0xFE;    /* Port3 Select: 11111110 = 0xFE */
+    P3OUT = 0x00;    /* Port3 Output: 00000000 = 0x00 */
+    P3DIR = 0xFF;    /* Port3 Direction: 11111111 = 0xFF */
 
 
-    // Port 4: CS
-    P4SEL = 0x00;    // Port4 Select: 00000000 = 0x00
-    P4OUT = 0x14;    // Port4 Output: 00010100 = 0x14
-    P4DIR = 0xFF;    // Port4 Direction: 11111111 = 0xFF
+    /* Port 4: CS */
+    P4SEL = 0x00;    /* Port4 Select: 00000000 = 0x00 */
+    P4OUT = 0x14;    /* Port4 Output: 00010100 = 0x14 */
+    P4DIR = 0xFF;    /* Port4 Direction: 11111111 = 0xFF */
 
-    // Port 5: SPI, LED
-    P5SEL = 0x0E;    // Port5 Select: 00001110 = 0x0E
-    P5OUT = 0x70;    // Port5 Output: 01110000 = 0x70
-    P5DIR = 0x70;    // Port5 Direction: 01110000 = 0x70
+    /* Port 5: SPI, LED */
+    P5SEL = 0x0E;    /* Port5 Select: 00001110 = 0x0E */
+    P5OUT = 0x70;    /* Port5 Output: 01110000 = 0x70 */
+    P5DIR = 0x70;    /* Port5 Direction: 01110000 = 0x70 */
 
 
-    P6SEL = 0xFF;    // Port6 Select: 11111111 = 0xFF
-    P6OUT = 0x00;    // Port6 Output: 00000000 = 0x00
-    P6DIR = 0xFF;    // Port6 Direction: 11111000 = 0xF8
+    P6SEL = 0xFF;    /* Port6 Select: 11111111 = 0xFF */
+    P6OUT = 0x00;    /* Port6 Output: 00000000 = 0x00 */
+    P6DIR = 0xFF;    /* Port6 Direction: 11111000 = 0xF8 */
 
 }
 
-void msp430_set_cpu_speed(uint32_t speed)
-{
-    irq_disable();
-    __msp430_cpu_speed = speed;
-    msp430_init_dco();
-    irq_enable();
-}
-
-/*---------------------------------------------------------------------------*/
-void msp430_init_dco(void)
+void msp430_init_cpuclk(uint32_t speed)
 {
-  /*----------------------- use external oszillator -------------------------*/
-  uint16_t i;
-
-  // Stop watchdog
-  WDTCTL = WDTPW + WDTHOLD;
-
-  BCSCTL1 = RSEL2;
-
-  // Wait for xtal to stabilize
-  do {
-    IFG1 &= ~OFIFG;                // Clear oscillator fault flag
-    for (i = 0xFF; i > 0; i--);    // Time for flag to set
-  }
-  while ((IFG1 & OFIFG) != 0);     // Oscillator fault flag still set?
-  switch (__msp430_cpu_speed) {
-  case MCLK_2MHZ_SCLK_1MHZ:
-      BCSCTL2  = (SELM_2 | DIVM_2) | (SELS | DIVS_3);
-      break;
-  case MCLK_4MHZ_SCLK_1MHZ:
-      BCSCTL2  = (SELM_2 | DIVM_1) | (SELS | DIVS_3);
-      break;
-  case MCLK_8MHZ_SCLK_1MHZ:
-      BCSCTL2 = SELM_2 | (SELS | DIVS_3);
-      break;
-  default:
-      BCSCTL2 = SELM_2 + SELS;     // MCLK and SMCLK = XT2 (safe)
-      break;
-  }
+    /* stop watchdog */
+    WDTCTL = WDTPW + WDTHOLD;
+    BCSCTL1 = RSEL2;
+
+    /* wait for XTAL to stabilize */
+    do {
+        /* clear oscillator fault flag */
+        IFG1 &= ~OFIFG;
+        /* time for flag to set */
+        for (uint16_t i = 0xFF; i > 0; i--) {}
+    }
+
+    /* oscillator fault flag still set? */
+    while ((IFG1 & OFIFG) != 0);
```

Comment:
Why are you writing the while loop here without a body, but the for-loop above you are writing with (empty) body. In general loops without bodies can lead to cross-platform problems and our [coding conventions state they should be *with* a body](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions#general)

Reply:
japp, this is a mistake here, will fix.